tests/common/osutil/test_nsbsd.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+from azurelinuxagent.common.utils.fileutil import read_file
+from azurelinuxagent.common.osutil.nsbsd import NSBSDOSUtil
+from .test_default import osutil_get_dhcp_pid_should_return_a_list_of_pids
+from tests.tools import *
+from os import path
+
+
+class TestNSBSDOSUtil(AgentTestCase):
+    dhclient_pid_file = "/var/run/dhclient.pid"
+
+    def setUp(self):
+        AgentTestCase.setUp(self)
+
+    def tearDown(self):
+        AgentTestCase.tearDown(self)
+
+    def test_get_dhcp_pid_should_return_a_list_of_pids(self):
```

Comment:
nsbsd gets the pid from a file, instead of looking at the running processes, so it needs its own unit tests